File: 1275/find-winner-on-a-tic-tac-toe-game.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        table = [[0]*3 for _ in range(3)]
        AB = -1 # [-1, +1] # A:-1, B:+1

        def testWin() -> bool:
            for i in range(3):
                A, B = 0, 0
                for j in range(3):
                    if table[i][j] == -1: A += 1
                    if table[i][j] == +1: B += 1
                if A==3 or B==3: return True

            for j in range(3):
                A, B = 0, 0
                for i in range(3):
                    if table[i][j] == -1: A += 1
                    if table[i][j] == +1: B += 1
                if A==3 or B==3: return True

            A, B = 0, 0
            for i in range(3):
                if table[i][i] == -1: A += 1
                if table[i][i] == +1: B += 1
            if A==3 or B==3: return True

            A, B = 0, 0
            for i in range(3):
                if table[i][2-i] == -1: A += 1
                if table[i][2-i] == +1: B += 1
            if A==3 or B==3: return True

            return False

        for m in moves:
            table[m[0]][m[1]] = AB
            if testWin():
                if AB == -1: return "A"
                else: return "B"
            AB *= -1
        if len(moves)==9:
            return "Draw"
        else:
            return "Pending"
# case 38/100: [[0,0],[1,1]]

File: 1275/find-winner-on-a-tic-tac-toe-game.c
char * tictactoe(int** moves, int movesSize, int* movesColSize){
    int table[3][3]={};
    int ans = 0; //no answer
    for(int i=0; i<movesSize; i++){
        int x = moves[i][0], y = moves[i][1];
        if(i%2==0) table[x][y] = 1;
        else table[x][y] = 2;

        if(table[0][0]==1 && table[0][1]==1 && table[0][2]==1) ans = 1;
        if(table[1][0]==1 && table[1][1]==1 && table[1][2]==1) ans = 1;
        if(table[2][0]==1 && table[2][1]==1 && table[2][2]==1) ans = 1;

        if(table[0][0]==1 && table[1][0]==1 && table[2][0]==1) ans = 1;
        if(table[0][1]==1 && table[1][1]==1 && table[2][1]==1) ans = 1;
        if(table[0][2]==1 && table[1][2]==1 && table[2][2]==1) ans = 1;

        if(table[0][0]==1 && table[1][1]==1 && table[2][2]==1) ans = 1;
        if(table[0][2]==1 && table[1][1]==1 && table[2][0]==1) ans = 1;

        if(table[0][0]==2 && table[0][1]==2 && table[0][2]==2) ans = 2;
        if(table[1][0]==2 && table[1][1]==2 && table[1][2]==2) ans = 2;
        if(table[2][0]==2 && table[2][1]==2 && table[2][2]==2) ans = 2;

        if(table[0][0]==2 && table[1][0]==2 && table[2][0]==2) ans = 2;
        if(table[0][1]==2 && table[1][1]==2 && table[2][1]==2) ans = 2;
        if(table[0][2]==2 && table[1][2]==2 && table[2][2]==2) ans = 2;

        if(table[0][0]==2 && table[1][1]==2 && table[2][2]==2) ans = 2;
        if(table[0][2]==2 && table[1][1]==2 && table[2][0]==2) ans = 2;

        if(ans==1) return "A";
        if(ans==2) return "B";
    }

    if(movesSize==9)return "Draw";
    else return "Pending";
}
